src/main/java/bio/terra/service/resourcemanagement/google/GoogleProjectService.java

```diff
@@ -404,56 +401,45 @@ public void updateIamPermissions(
 
     GetIamPolicyRequest getIamPolicyRequest = new GetIamPolicyRequest();
 
-    Exception lastException = null;
-    int retryWait = INITIAL_WAIT_SECONDS;
-    for (int i = 0; i < RETRIES; i++) {
-      try {
-        CloudResourceManager resourceManager = cloudResourceManager();
-        Policy policy =
-            resourceManager.projects().getIamPolicy(projectId, getIamPolicyRequest).execute();
-        final List<Binding> bindingsList = policy.getBindings();
-
-        switch (permissionOp) {
-          case ENABLE_PERMISSIONS:
-            for (Map.Entry<String, List<String>> entry : userPermissions.entrySet()) {
-              Binding binding = new Binding().setRole(entry.getKey()).setMembers(entry.getValue());
-              bindingsList.add(binding);
-            }
-            break;
-
-          case REVOKE_PERMISSIONS:
-            // Remove members from the current policies
-            for (Map.Entry<String, List<String>> entry : userPermissions.entrySet()) {
-              CollectionUtils.filter(
-                  bindingsList,
-                  b -> {
-                    if (Objects.equals(b.getRole(), entry.getKey())) {
-                      // Remove the members that were passed in
-                      b.setMembers(ListUtils.subtract(b.getMembers(), entry.getValue()));
-                      // Remove any entries from the bindings list with no members
-                      return !b.getMembers().isEmpty();
-                    }
-                    return true;
-                  });
-            }
-        }
-
-        policy.setBindings(bindingsList);
-        SetIamPolicyRequest setIamPolicyRequest = new SetIamPolicyRequest().setPolicy(policy);
-        resourceManager.projects().setIamPolicy(projectId, setIamPolicyRequest).execute();
-        return;
-      } catch (IOException | GeneralSecurityException ex) {
-        logger.info("Failed to enable iam permissions. Retry " + i + " of " + RETRIES, ex);
-        lastException = ex;
-      }
-
-      TimeUnit.SECONDS.sleep(retryWait);
-      retryWait = retryWait + retryWait;
-      if (retryWait > MAX_WAIT_SECONDS) {
-        retryWait = MAX_WAIT_SECONDS;
-      }
-    }
-    throw new UpdatePermissionsFailedException("Cannot update iam permissions", lastException);
+    Callable<Void> retryable =
+        () -> {
+          CloudResourceManager resourceManager = cloudResourceManager();
+          Policy policy =
+              resourceManager.projects().getIamPolicy(projectId, getIamPolicyRequest).execute();
+          final List<Binding> bindingsList = policy.getBindings();
+
+          switch (permissionOp) {
+            case ENABLE_PERMISSIONS:
+              for (Map.Entry<String, List<String>> entry : userPermissions.entrySet()) {
```

Comment:
This a case where `var` can help IMO
```suggestion
              for (var entry : userPermissions.entrySet()) {
```